src/app/domain/authentication/services/AuthService.ts

```diff
@@ -0,0 +1,64 @@
+import { EventEmitter } from 'events'
+
+import Minilog from '@cozy/minilog'
+
+import type CozyClient from 'cozy-client'
+
+import { asyncLogout } from '/libs/intents/localMethods'
+import { safePromise } from '/utils/safePromise'
+
+export const authLogger = Minilog('[AuthService]')
+const emitter = new EventEmitter()
+
+let userRevoked = false
+let clientInstance: CozyClient | null = null
+
+// Setter
+export const setUserRevoked = (status: boolean): void => {
+  userRevoked = status
+  authLogger.info(`User revoked set to "${String(status)}"`)
+
+  if (userRevoked) emitter.emit('userRevoked', userRevoked)
+  if (!userRevoked) safePromise(handleOnUnrevoked)()
+}
+
+// Getter
+export const isUserRevoked = (): boolean => userRevoked
+
+// Handle token error
+const handleTokenError = (): void => {
+  authLogger.warn('Token revoked')
+  setUserRevoked(true)
+}
+
+// Handle user confirmation of the error
+const handleOnUnrevoked = async (): Promise<void> => {
+  try {
+    if (clientInstance === null) throw new Error('No client instance set')
+    authLogger.info('User confirmed the error, logging out')
+    await asyncLogout(clientInstance)
```

Comment:
`asyncLogout` will call 
```
await sendKonnectorsLogs(client)
await client.logout()
```

Both those method will do HTTP requests to the stack and will fail if client is revoked.

The process may correctly catch those exceptions, but we will still get the `log.error` calls into sentry. Maybe we should just clean the asyncstorage ?

Reply:
Agree, I think we should call everything that doesn't rely on a client, will look into it
- [ ] Updated